src/main/java/seedu/apollo/task/Event.java

```diff
@@ -17,32 +23,68 @@ public class Event extends Task {
     protected LocalDateTime to;
 
+
     /**
      * Initialises as in Task, with added parsing for start and end dates.
      * If parsing is not possible, save date(s) as String(s).
      *
      * @param description String describing the Task.
-     * @param fromString String describing the start date.
-     * @param toString String describing the end date.
+     * @param fromString  String describing the start date.
+     * @param toString    String describing the end date.
      * @throws DateTimeParseException If either date is not entered in right format.
-     * @throws DateOrderException If end date occurs before the start date.
-     * @throws DateOverException If start date occurs before the current date.
+     * @throws DateOrderException     If end date occurs before the start date.
+     * @throws DateOverException      If start date occurs before the current date.
      */
-    public Event(String description, String fromString, String toString)
-            throws DateTimeParseException, DateOrderException, DateOverException {
+    public Event(String description, String fromString, String toString, TaskList taskList, ModuleList moduleList)
+            throws DateTimeParseException, DateOrderException, DateOverException,
+            EventEventClashException, EventModuleClashException {
         super(description);
         this.from = LocalDateTime.parse(fromString);
         this.to = LocalDateTime.parse(toString);
 
         if (from.isAfter(to)) {
             throw new DateOrderException();
         }
+        if (clashTaskList(to, from, taskList)) {
+            throw new EventEventClashException();
+        }
+        if (clashModuleList(to, from, moduleList)) {
+
+            throw new EventModuleClashException();
+        }
+
 
         if (from.isBefore(LocalDateTime.now())) {
             throw new DateOverException(getType(), description, null, from, to);
         }
     }
 
+    public boolean clashTaskList(LocalDateTime to, LocalDateTime from, TaskList tasksList) {
+        for (Task task : tasksList) {
+            if (task instanceof Event) {
+                Event event = (Event) task;
+                if (from.isBefore(event.to) || to.isAfter(event.from)) {
+                    return true; // Overlap detected
+                }
+            }
+        }
+        return false;
+    }
+
+    public boolean clashModuleList(LocalDateTime to, LocalDateTime from, ModuleList moduleList) {
+        for (Module module : moduleList) {
+            ArrayList<Timetable> timetableList = module.getModuleTimetable();
+            for (Timetable timetable : timetableList) {
+                LocalDateTime start = LocalDateTime.parse(timetable.getStartTime());
+                LocalDateTime end = LocalDateTime.parse(timetable.getEndTime());
+                if (from.isBefore(end) || to.isAfter(start)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
```

Comment:
Perhaps you could reduce the level of nesting in these 2 methods to make it more readable.